Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -334,14 +338,17 @@ JPEGImageIO::ReadImageInformation()
     case 1:
       m_PixelType = IOPixelEnum::SCALAR;
       break;
-    case 2:
-      m_PixelType = IOPixelEnum::VECTOR;
-      break;
     case 3:
       m_PixelType = IOPixelEnum::RGB;
       break;
     case 4:
+      // TODO
       m_PixelType = IOPixelEnum::RGBA;
+      itkWarningMacro("JPEG image may be opened incorrectly");
```

Comment:
4 components JPEGs are unlikely, probably never, RGBA. Mostly CMYK, rare, here is one example. Opened with this IO it looks more "pale" as in other viewers. Looks the same if opened as RBG, BTW. More work for correct conversion from CMYK is required (example in e.g. Gimp's jpeg-load.c) and will be very rarely used.

Reply:
Here is how Gimp handles this
```
  switch (cinfo.output_components)
    {
    case 1:
      image_type = GIMP_GRAY;
      layer_type = GIMP_GRAY_IMAGE;
      break;

    case 3:
      image_type = GIMP_RGB;
      layer_type = GIMP_RGB_IMAGE;
      break;

    case 4:
      if (cinfo.out_color_space == JCS_CMYK)
        {
          image_type = GIMP_RGB;
          layer_type = GIMP_RGB_IMAGE;
          break;
        }
      /*fallthrough*/

    default:
      g_message ("Don't know how to load JPEG images "
                 "with %d color channels, using colorspace %d (%d).",
                 cinfo.output_components, cinfo.out_color_space,
                 cinfo.jpeg_color_space);
      return -1;
      break;
    }
```

Reply:
CMYK is already done, will be added in separate PR. The files are not unusual, Photoshop has the option to export CMYK jpegs. Current "RGBA" part is bug.